arch/risc-v/src/k210/k210_head.S

```diff
@@ -215,17 +215,17 @@ normal_irq:
  *  Name: g_intstackalloc and g_intstackbase
  ************************************************************************************/
 
-#if CONFIG_ARCH_INTERRUPTSTACK > 7
+#if CONFIG_ARCH_INTERRUPTSTACK > 15
   .bss
-  .align  8
+  .align  16
   .global g_intstackalloc
   .global g_intstackbase
   .type   g_intstackalloc, object
   .type   g_intstackbase, object
 g_intstackalloc:
```

Comment:
I believe we should first backport this commit 7ce1033aa2 (from PR https://github.com/apache/incubator-nuttx/pull/3518 @masayuki2009 ).
Otherwise the SMP mode for K210 will still fail.

Reply:
My understanding is that @masayuki2009 change was required _only_ after @xiaoxiang781216 change.  Is that correct?  If so we don't need to backport it.

BTW, the `.align` here also should be `.balign`.

Reply:
> My understanding is that @masayuki2009 change was required only after @xiaoxiang781216 change. Is that correct? If so we don't need to backport it.

I don't think so, because @masayuki2009's commit earlier than @xiaoxiang781216 stack refactor.

Reply:
Oh, the "recent refactoring" on @masayuki2009's PR might refer to this commit 3f67c67aaf from @xiaoxiang781216, which we also need to evaluate backporting it.
It seems to be self-contained, not dependent on the big stack rework.

Reply:
3f67c67 is also rather large and recent, I think we'd better leave it for the next one.

If @masayuki2009's PR is independent of that change and fixes a bug (that exists in releases/10.1), then we should backport it.

Reply:
Sorry for the late reply but we need to apply similar logic to allocate stack for SMP as I did in https://github.com/apache/incubator-nuttx/pull/3518/files

Reply:
@masayuki2009 So we might need to backport both commits from https://github.com/apache/incubator-nuttx/pull/3636

Reply:
> @masayuki2009 So we might need to backport both commits from #3636

@gustavonihei Yes, please.